data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncing.java

```diff
@@ -85,10 +91,13 @@ static class SyncStatusData {
     private final UInt64 currentSlot;
     private final UInt64 slotsBehind;
 
-    public SyncStatusData(final SyncDataProvider syncProvider) {
+    public SyncStatusData(
+        final SyncDataProvider syncProvider,
+        ExecutionClientDataProvider executionClientDataProvider) {
       final SyncingStatus status = syncProvider.getSyncingStatus();
       final SyncState syncState = syncProvider.getCurrentSyncState();
-      this.isSyncing = !syncState.isInSync();
+      this.isSyncing =
+          syncState.isSyncing() || !executionClientDataProvider.isExecutionClientAvailable();
```

Comment:
isSyncing already pulls in Optimistic_syncing and awaiting EL..., so the ELDataProvider is basically here because we don't update the syncing state properly? Maybe this is just fixing a symptom that is a problem elsewhere if that's the case?
My main concerns:
 - we seem to be overriding the results of our own SyncService with extra context, indicating maybe something is wrong with the syncService
 - this is extremely close to an api owning business logic, it does seem to blur the line a lot... I'd prefer the api functions aren't applying special business logic unless we understand really well why we need it, and potentially even then the actual answer may be adjusting the flow, not business logic in the api...

Reply:
IMO, it depends on what the `SyncService` is supposed to do. Currently it doesn't consider the EL status (which is not necessarily incorrect). I agree with the extreme closeness of including business logic in the API and that it looks like there is something wrong with the `SyncService`.
I will give it another try and see how the EL availability could be taken into consideration directly in the `SyncService`